docs/source/cuda/cudapysupported.rst

```diff
@@ -0,0 +1,202 @@
+========================================
+Supported Python features in CUDA Python
+========================================
+
+This page lists the Python features supported in the CUDA Python.  This includes
+all kernel and device functions compiled with ``@cuda.jit`` and other higher
+level Numba decorators that targets the CUDA GPU.
+
+Language
+========
+
+Execution Model
+---------------
+
+CUDA Python maps directly to the *single-instruction multiple-thread*
+execution (SIMT) model of CUDA.  Each instruction is implicitly
+executed by multiple threads in parallel.  With this execution model, array
+expression is less useful because we don't want multiple threads to perform
+the same task.  Instead, we want threads to perform a task in a cooperative
+fashion.
+
+The CUDA execution model also has some limitations.
+For instance, dynamic memory allocation is limited due to the
```

Comment:
Thinking a little more, I think it would probably be simpler if this part just said "The CUDA execution model also has some limitations. Features that require dynamic memory allocation are not supported", without going into the discussion about the heap size and where memory would be allocated for it.